src/components/CountryTimeChart.vue

```diff
@@ -0,0 +1,123 @@
+
+    private yearScale: d3.ScaleLinear<number, number> | null = null;
+    private anemiaScale: d3.ScaleLinear<number, number> | null = null;
+    private gdpScale: d3.ScaleLinear<number, number> | null = null;
+
+    @Watch('data')
+    @Watch('yearRange')
+    public createSVG() {
+      if (this.data && this.years) {
+        this.yearScale = d3.scaleLinear()
+          .domain([Math.min(...this.years), Math.max(...this.years)])
+          .range([CountryTimeChart.PADDING, CountryTimeChart.WIDTH - CountryTimeChart.PADDING]);
+
+        const anemiaValues = this.data
+          .flatMap(d => d.values.filter(p => p['Item Code'] === ItemCodes.ANEMIA_CODE))
+          .map(p => Number.parseFloat(p.Value));
+        this.anemiaScale = d3.scaleLinear()
+          .domain([Math.min(...anemiaValues), Math.max(...anemiaValues)])
+          .range([CountryTimeChart.HEIGHT - CountryTimeChart.PADDING, CountryTimeChart.PADDING]);
+
+        const gdpValues = this.data
+          .flatMap(d => d.values.filter(p => p['Item Code'] === ItemCodes.GDP_CODE))
+          .map(p => Number.parseFloat(p.Value));
+        this.gdpScale = d3.scaleLinear()
+          .domain([Math.min(...gdpValues), Math.max(...gdpValues)])
+          .range([CountryTimeChart.HEIGHT - CountryTimeChart.PADDING, CountryTimeChart.PADDING]);
+
+        const svg = d3
+          .select('svg.countrytimechart')
+          .attr('width', CountryTimeChart.WIDTH)
+          .attr('height', CountryTimeChart.HEIGHT);
+
+        svg.selectAll('g').remove();
+        const svgGroup = svg.append('g');
+
+        svgGroup.append('g')
+          .call(d3.axisLeft(this.anemiaScale))
+          .attr('transform', `translate(${CountryTimeChart.PADDING}, 0)`)
+          .style('color', '#F55');
+        svgGroup.append('g')
+          .call(d3.axisRight(this.gdpScale))
+          .attr('transform', `translate(${CountryTimeChart.WIDTH - CountryTimeChart.PADDING}, 0)`)
+          .style('color', '#55F');
+        svgGroup.append('g')
+          .call(d3.axisBottom(this.yearScale))
+          .attr('transform', `translate(0, ${CountryTimeChart.HEIGHT - CountryTimeChart.PADDING})`);
+
+        svgGroup.append('text')
+          .text('Year')
+          .attr('y', CountryTimeChart.HEIGHT - 20)
+          .attr('x', 50)
+          .style('font-size', '10px');
+
+        svgGroup.append('text')
+          .text('Prevalence of anemia among women of reproductive age (15-49 years)')
+          .attr('transform', `translate(15, ${CountryTimeChart.HEIGHT - CountryTimeChart.PADDING}),rotate(-90)`)
+          .style('font-size', '10px');
+
+        svgGroup.append('text')
+          .text('Gross domestic product per capita ($)')
+          .attr('transform',
+            `translate(${CountryTimeChart.WIDTH}, ${CountryTimeChart.HEIGHT - CountryTimeChart.PADDING}),rotate(-90)`)
+          .style('font-size', '10px');
+
+        const anemiaLine = d3.line<DataPoint>()
+          .x(d => this.yearScale(d.Year))
```

Comment:
Die scale hier und in den nächsten Zeilen ist possibly null, da gibt's nen Error